astropy/coordinates/attributes.py

```diff
@@ -444,6 +444,8 @@ def convert_input(self, value):
 
         if value is None:
             return None, False
+        elif isinstance(value, SkyCoord) and isinstance(value.frame, self._frame):
+            return value, False
```

Comment:
I don't quite recall from when this was created, but the docstring above suggests a `SkyCoord` could be input, but the frame would be used. This is consistent with what happens below, so I think we need to return `value.frame` here.

Reply:
Aren't you confusing `FrameAttribute` with `CoordinateAttribute` here? Would returning `value.frame` keep the representation data?

Reply:
Ah, ok, yes it would. Ok.

I am still confused what's with the distinction of `SkyCoord` and `Frame` with data inside...

Reply:
OK, glad it is resolved. The way I usually remember it is that `SkyCoord` can have attributes of any frame, like a `location` (which ICRS could not carry, but is needed to transform to `AltAz`).